### PR TITLE
add migrate, randomkey, restore and wait keys

### PR DESCRIPTION
**Pull request by mikail-khan**

### CODE REVIEW

File: src/main/scala/zio/redis/api/Keys.scala [hunk 1]
```diff
@@ -29,4 +57,5 @@ trait Keys {
   final val ttl    = RedisCommand("TTL", StringInput, DurationOutput)
   final val typeOf = RedisCommand("TYPE", StringInput, StringOutput)
   final val unlink = RedisCommand("UNLINK", NonEmptyList(StringInput), LongOutput)
+  final val wait_  = RedisCommand("WAIT", Tuple2(LongInput, LongInput), LongOutput)
```
Comment (mikail-khan) [inline src/main/scala/zio/redis/api/Keys.scala:60]: appended an underscore until we know what to do with this

File: src/main/scala/zio/redis/options/Shared.scala [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package zio.redis.options
 
 trait Shared {
+  sealed case class Limit(offset: Int, count: Int)
```
Comment (mikail-khan) [inline src/main/scala/zio/redis/options/Shared.scala:4]: shared by `SortedSets` and `Keys`

Comment (mijicd) [reply src/main/scala/zio/redis/options/Shared.scala:4]: It doesn't seem to be used in keys though.

Comment (mikail-khan) [reply src/main/scala/zio/redis/options/Shared.scala:4]: I made that change to implement the `sort` command but then realised that `sort` can't be introduced yet until the current design can accommodate it. I can move it back to `SortedSets` if `sort` is not viable.

Comment (mijicd) [reply src/main/scala/zio/redis/options/Shared.scala:4]: Already did it on interpreter.

**Pull request merged**